FlickTrainer/app/src/main/java/jp/co/azz/flicktrainer/databases/HistoryDao.java

```diff
@@ -0,0 +1,200 @@
+     * @return {void}
+     */
+    public void insertHistory(int score, int inputStringPerMinute, int timesOfError, String errorString) {
+
+        //システム日付の設定（出力フォーマット：YY/MM/DD HH:MM:SS）
+        Date date = new Date();
+        SimpleDateFormat sdf = new SimpleDateFormat("yy/MM/dd HH:mm:ss");
+        String playDate = sdf.format(date);
+
+        //Historyテーブル INSERT文
+        ContentValues cv = new ContentValues();
+        cv.put(DatabaseContract.History.COLUMN_NAME_SCORE, score);
+        cv.put(DatabaseContract.History.COLUMN_NAME_INPUT_STRING_PER_MINUTES, inputStringPerMinute);
+        cv.put(DatabaseContract.History.COLUMN_NAME_TIMES_OF_ERROR, timesOfError);
+        cv.put(DatabaseContract.History.COLUMN_NAME_ERROR_STRING, errorString);
+        cv.put(DatabaseContract.History.COLUMN_NAME_PLAY_DATE, playDate);
+
+        //Insert処理
+        SQLiteDatabase db = dataBaseHelper.getWritableDatabase();
+        db.insert(DatabaseContract.History.TABLE_NAME, null, cv);
+
+    }
+
+    /**
+     * Historyテーブル に文をDELETE発行するメソッド （全件クリア）
+     *
+     * @return {void}
+     */
+    public void deleteHistoryAll() {
+        //historyテーブル DELETE処理
+        SQLiteDatabase db = dataBaseHelper.getWritableDatabase();
+        db.delete(DatabaseContract.History.TABLE_NAME, null, null);
+
+    }
+
+    /**
+     * Historyテーブル にDELETE文を発行するメソッド (一番古いPlayDateを削除)
+     *
+     * @return {void}
+     */
+    public void deleteHistoryOldPlayDate() {
+        //historyテーブル DELETE文
+        SQLiteDatabase db = dataBaseHelper.getWritableDatabase();
+        db.execSQL("DELETE FROM " + DatabaseContract.History.TABLE_NAME +
+                " WHERE " + DatabaseContract.History.COLUMN_NAME_PLAY_DATE + " = " +
+                " ( SELECT MIN( " + DatabaseContract.History.COLUMN_NAME_PLAY_DATE + " ) FROM " +
+                DatabaseContract.History.TABLE_NAME + " ) ");
+    }
+
+    /**
+     * Userテーブル にselect文を発行するメソッド（練習回数を取得）
+     *
+     * @return {int} 練習回数
+     */
+    public int selectUser() {
+        //初期化処理
+        int timesOfPractice = 0;
+        //Userテーブル SELECT 文発行
+        try (
+                SQLiteDatabase db = dataBaseHelper.getReadableDatabase();
+                Cursor cursor = db.query(
+                        DatabaseContract.User.TABLE_NAME,
+                        new String[]{DatabaseContract.User.COLUMN_TIMES_OF_PRACTICE},
+                        null,
+                        null,
+                        null,
+                        null,
+                        null
+                );
+        ) {
+            while (cursor.moveToNext()) {
+                timesOfPractice = cursor.getInt(cursor.getColumnIndex(DatabaseContract.User.COLUMN_TIMES_OF_PRACTICE));
+            }
+        }
+        //取得した練習回数を戻す
+        return timesOfPractice;
+    }
+
+    /**
+     * Userテーブルの UPDATE文を発行するメソッド（プレイ回数を1回ずつ追加する）
+     *
+     * @return {void}
+     */
+    public void updateUser() {
```

Comment:
Userようのファイルに分けた上で修正してください。
```suggestion
    public void update() {
```

Reply:
メソッド名をupdateに変更しました。